src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,163 @@
-
+        if (table == null || table.length == 0) {
+            table = transfer();
+        }
+        Node<K, V> first;
+        if ((first = table[getIndex(key)]) == null) {
+            table[getIndex(key)] = new Node<>(hash(key), key, value, null);
+        } else {
+            if (first.hash == hash(key)
+                    && (first.key == key || (key != null && key.equals(first.key)))) {
+                first.value = value;
+                return;
+            }
+            Node<K, V> e;
+            Node<K, V> tail;
+            if ((e = first.next) != null) {
+                do {
+                    tail = e;
+                    if (e.hash == hash(key)
+                            && (e.key == key || (key != null && key.equals(e.key)))) {
+                        e.value = value;
+                        return;
+                    }
+                } while ((e = e.next) != null);
+                tail.next = new Node<>(hash(key), key, value, null);
+            } else {
+                first.next = new Node<>(hash(key), key, value, null);
+            }
+        }
+        size++;
+        if (size > threshold) {
+            transfer();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : key.hashCode();
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> first;
+        if (table != null && (table.length) > 0 && ((first = table[getIndex(key)]) != null)) {
+            if (first.hash == hash(key)
+                    && (first.key == key || (key != null && key.equals(first.key)))) {
+                return first;
+            }
+            Node<K, V> e;
+            if ((e = first.next) != null) {
+                do {
+                    if (e.hash == hash(key)
+                            && (e.key == key || (key != null && key.equals(e.key)))) {
+                        return e;
+                    }
+                } while ((e = e.next) != null);
+            }
+        }
+        return null;
+    }
+
+    private int getIndex(K key) {
+        return hash(key) & (table.length - 1);
+    }
+
+    private int resize() {
+        int oldCap = (table == null) ? 0 : table.length;
+        int oldThr = threshold;
+        int newCap;
+        int newThr;
+        if (oldCap == 0) {
+            newCap = DEFAULT_INITIAL_CAPACITY;
+            newThr = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        } else {
+            newCap = oldCap << 1;
+            newThr = oldThr << 1;
+        }
+        threshold = newThr;
+        return newCap;
+    }
+
+    private Node<K, V>[] transfer() {
+        Node<K, V>[] oldTab = table;
+        table = (Node<K, V>[]) new Node[resize()];
```

Comment:
```suggestion
        table = (Node<K, V>[]) new Node[table.length * 2];
```
make a constant from 2